zerver/lib/actions.py

```diff
@@ -4985,23 +4986,42 @@ def get_status_dict(requesting_user_profile: UserProfile) -> Dict[str, Dict[str,
     return UserPresence.get_status_dict_by_realm(requesting_user_profile.realm_id)
 
 def get_cross_realm_dicts() -> List[Dict[str, Any]]:
+    # Using the deferred import to avoid circular import error.
+    from zerver.lib.events import format_user_row
+
     users = bulk_get_users(list(settings.CROSS_REALM_BOT_EMAILS), None,
                            base_query=UserProfile.objects.filter(
-                               realm__string_id=settings.SYSTEM_BOT_REALM)).values()
-    return [{'email': user.email,
-             'user_id': user.id,
-             'is_admin': user.is_realm_admin,
-             'is_bot': user.is_bot,
-             'avatar_url': avatar_url(user),
-             'timezone': user.timezone,
-             'date_joined': user.date_joined.isoformat(),
-             'bot_owner_id': None,
-             'full_name': user.full_name}
-            for user in users
-            # Important: We filter here, is addition to in
-            # `base_query`, because of how bulk_get_users shares its
-            # cache with other UserProfile caches.
-            if user.realm.string_id == settings.SYSTEM_BOT_REALM]
+                           realm__string_id=settings.SYSTEM_BOT_REALM)).values()
+    result = []
+    for user in users:
+        # Important: We filter here, is addition to in
+        # `base_query`, because of how bulk_get_users shares its
+        # cache with other UserProfile caches.
+        if user.realm.string_id == settings.SYSTEM_BOT_REALM:
```

Comment:
Unnecessary if conditional? We're already filtering for this in line 4994.

Reply:
I think it's necessary .The comment addresses the second conditional .

Reply:
Oh, you're right. My mistake, sorry about that.

Reply:
Should I let the comment be there above the if conditional ?

Reply:
Nah, it can be left where it is now. It's in the right place. I should have read the comment a little more carefully.